src/listener/mod.rs

```diff
@@ -0,0 +1,18 @@
+mod common;
```

Comment:
there are no "common" or "utils" allowed in projects ever!
this shows we did not carefully consider where to place these methods/traits etc.
It is confusing to the developer and no one can guess what this holds as the name is not meaningful.
better names can be "os_generic" or "base_listener" or "base_keyboard_listener" etc.